airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultDiscoverCatalogWorker.java

```diff
@@ -56,16 +65,20 @@ public AirbyteCatalog run(final StandardDiscoverCatalogInput discoverSchemaInput
 
       LineGobbler.gobble(process.getErrorStream(), LOGGER::error);
 
-      final Optional<AirbyteCatalog> catalog;
+      final Map<Type, List<AirbyteMessage>> messagesByType;
+
       try (final InputStream stdout = process.getInputStream()) {
-        catalog = streamFactory.create(IOs.newBufferedReader(stdout))
-            .filter(message -> message.getType() == Type.CATALOG)
-            .map(AirbyteMessage::getCatalog)
-            .findFirst();
+        messagesByType = streamFactory.create(IOs.newBufferedReader(stdout))
+            .collect(Collectors.groupingBy(AirbyteMessage::getType));
```

Comment:
This is how we are building failure reasons from trace messages for each of these jobs:
- Instead of just filtering for the desired output when reading the stream initially, collect the messages by type
- filter for our desired output
- filter for a trace message if needed and use the existing FailureHelper to produce a FailureReason from it